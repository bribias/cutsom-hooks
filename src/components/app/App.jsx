import React from 'react';
import { BrowserRouter, Switch, Route, Link } from 'react-router-dom';import Character from '../characters/Character';
import HeyArnoldDetailsContainer from '../../containers/HeyArnoldDetailsContainer';
import HeyArnoldContainer from '../../containers/HeyArnoldContainer';

export default function App() {
  return (
    <BrowserRouter>
      <header>
        <nav>
          <Link to="/">
            Home
          </Link>
        </nav>
      </header>
      <Switch>
        <Route path="/:id">
          <HeyArnoldDetailsContainer />
        </Route>
        <Route path="/">
          <HeyArnoldContainer />
        </Route>
      </Switch>
    </BrowserRouter>
  );
}
