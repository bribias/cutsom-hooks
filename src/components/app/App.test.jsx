import React from 'react';
import { render, screen } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { MemoryRouter } from 'react-router-dom';
import App from './App';
import listData from '../../fixtures/listData.json';
import detailData from '../../fixtures/detailData.json';

const server = setupServer(
    rest.get('https://hey-arnold-api.herokuapp.com/api/v1/characters', (req, res, ctx) => {
        return res(ctx.json(listData));
    })
);

describe('Hey Arnold App', () => {
    beforeAll(() => server.listen());
    afterAll(() => server.close());

    it('displays a list of hey arnold characters', async () => {
        const { container } = render(
            <MemoryRouter>
                <App />
            </MemoryRouter>
        );

        screen.getAllByAltText('loading spinner');

        const ul = await screen.findByRole('list', { name: 'character-list' });
        expect(ul).not.toBeEmptyDOMElement();
        expect(container).toMatchSnapshot();
    });
});