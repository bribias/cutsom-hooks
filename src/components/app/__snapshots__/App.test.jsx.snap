// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hey Arnold App displays a list of hey arnold characters 1`] = `
<div>
  <header>
    <nav>
      <a
        href="/"
      >
        Home
      </a>
    </nav>
  </header>
  <ul
    aria-label="character-list"
  >
    <a
      href="/5da237699734fdcb7bef8f5c"
    >
      <li>
        <button>
          <img
            alt="Nancy"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
          />
          <h2>
            Nancy
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f59"
    >
      <li>
        <button>
          <img
            alt="Grandpa's grandfather"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f7/Grandpa%27s_grandfather.png/revision/latest/scale-to-width-down/310?cb=20190228081015"
          />
          <h2>
            Grandpa's grandfather
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f52"
    >
      <li>
        <button>
          <img
            alt="Grandpa Phil"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
          />
          <h2>
            Grandpa Phil
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f58"
    >
      <li>
        <button>
          <img
            alt="Grandpa's father"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e0/Grandpa%27s_father.png/revision/latest/scale-to-width-down/310?cb=20190228081112"
          />
          <h2>
            Grandpa's father
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f70"
    >
      <li>
        <button>
          <img
            alt="Phoebe Heyerdahl"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/9/93/Phoebe_%28classic%29.png/revision/latest/scale-to-width-down/247?cb=20190816022206"
          />
          <h2>
            Phoebe Heyerdahl
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f7e"
    >
      <li>
        <button>
          <img
            alt="Eugene Horowitz"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/9/99/EAFDB121-EC13-4F12-849E-37037A95E3D6.png/revision/latest/scale-to-width-down/270?cb=20190816022130"
          />
          <h2>
            Eugene Horowitz
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f82"
    >
      <li>
        <button>
          <img
            alt="Ray"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/3/34/Ray.png/revision/latest/scale-to-width-down/310?cb=20181006091830"
          />
          <h2>
            Ray
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f8b"
    >
      <li>
        <button>
          <img
            alt="Shelley"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256"
          />
          <h2>
            Shelley
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f76"
    >
      <li>
        <button>
          <img
            alt="Lila Sawyer"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/b/b0/Lila_Sawyer.png/revision/latest/scale-to-width-down/310?cb=20171212053404"
          />
          <h2>
            Lila Sawyer
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f7c"
    >
      <li>
        <button>
          <img
            alt="Uncle Stinky"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/8/89/Uncle_Stinky.jpg/revision/latest/scale-to-width-down/310?cb=20120924204406"
          />
          <h2>
            Uncle Stinky
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f55"
    >
      <li>
        <button>
          <img
            alt="Arnie"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/4/42/Arnie.jpg/revision/latest/scale-to-width-down/310?cb=20110109195825"
          />
          <h2>
            Arnie
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f66"
    >
      <li>
        <button>
          <img
            alt="Olga Pataki"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/5/5e/Olga_Pataki.png/revision/latest/scale-to-width-down/310?cb=20180124053913"
          />
          <h2>
            Olga Pataki
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f6b"
    >
      <li>
        <button>
          <img
            alt="Timberly Johanssen"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e8/Timberly.png/revision/latest/scale-to-width-down/310?cb=20171210045406"
          />
          <h2>
            Timberly Johanssen
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f6d"
    >
      <li>
        <button>
          <img
            alt="Jerry Berman"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/2/2f/Jerry_Berman.png/revision/latest/scale-to-width-down/310?cb=20171219061306"
          />
          <h2>
            Jerry Berman
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f73"
    >
      <li>
        <button>
          <img
            alt="Rhonda Wellington Lloyd"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/b/b4/C1BBB448-FE38-4F7B-98FA-2137F277F9C5.png/revision/latest/scale-to-width-down/213?cb=20190816022649"
          />
          <h2>
            Rhonda Wellington Lloyd
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f78"
    >
      <li>
        <button>
          <img
            alt="Stinky Peterson"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/2/24/Stinky_Peterson.png/revision/latest/scale-to-width-down/310?cb=20160516072820"
          />
          <h2>
            Stinky Peterson
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f71"
    >
      <li>
        <button>
          <img
            alt="Kyo Heyerdahl"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cf/Kyo_Heyerdahl.png/revision/latest/scale-to-width-down/310?cb=20190310221807"
          />
          <h2>
            Kyo Heyerdahl
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f8e"
    >
      <li>
        <button>
          <img
            alt="Iggy"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/0/0e/Iggy.png/revision/latest/scale-to-width-down/310?cb=20190210045957"
          />
          <h2>
            Iggy
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f77"
    >
      <li>
        <button>
          <img
            alt="Mr. Sawyer"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/4/45/Mr._Sawyer.png/revision/latest/scale-to-width-down/310?cb=20171219055204"
          />
          <h2>
            Mr. Sawyer
          </h2>
        </button>
      </li>
    </a>
    <a
      href="/5da237699734fdcb7bef8f84"
    >
      <li>
        <button>
          <img
            alt="Thaddeus \\"Curly\\" Gammelthorpe"
            height="200px"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/b/ba/Classic_Curly_promo.png/revision/latest/scale-to-width-down/246?cb=20181010011614"
          />
          <h2>
            Thaddeus "Curly" Gammelthorpe
          </h2>
        </button>
      </li>
    </a>
  </ul>
</div>
`;
